### ISSUE TITLE
Unexpected failure in foreach statement

### ISSUE THREAD

**Issue by danielbachhuber** [sirbrillig/phpcs-variable-analysis#72]
```php
foreach ( $recipe_json as $k => $v ) {
	if ( false !== strpos( $k, '_rendered' ) ) {
		unset( $recipe_json[ $k ] );
	}
}
```

In this statement, `$v` is flagged as unused. I guess this is the same as #65?

**Comment by sirbrillig** [comment]
It should be the same, yes. If you set `allowUnusedForeachVariables` to `true`, then it will not be reported. But this does continue that discussion, specifically

> we can consider if we want that option to be on by default

What do you think? It seems like there's an obvious benefit, but the reason I haven't enabled it by default already is that I haven't taken the time to examine the possible downsides. It definitely hides unused variables and I wonder if that's something that could be problematic for some code.

**Comment by danielbachhuber** [comment]
It'd make sense to me to enable `allowUnusedForeachVariables` by default because PHP doesn't offer any alternatives.

Similarly, I also had to enable `allowUnusedCaughtExceptions`.

**Comment by sirbrillig** [comment]
> PHP doesn't offer any alternatives

I mean, there's `foreach ( array_keys( $recipe_json ) as $key )`, but I see your point. For large arrays that's another time through.

Let me work up a PR and we'll see what that looks like.

> Similarly, I also had to enable `allowUnusedCaughtExceptions`.

That's a good point and not one I've seen mentioned a lot, but it could probably help more people to have that on be default as well.

**Comment by danielbachhuber** [comment]
> > PHP doesn't offer any alternatives
> 
> I mean, there's `foreach ( array_keys( $recipe_json ) as $key )`, but I see your point. For large arrays that's another time through.

Ah, good point. Even so, I think this is one of those border areas (i.e. there's a reasonable explanation for an unused variable).
